src/smdba/postgresqlgate.py

```diff
@@ -801,7 +801,11 @@ def _perform_enable_backups(self, **args):
             b_dir_temp = os.path.join(backup_dir, 'tmp')
             cwd = os.getcwd()
             os.chdir(self.config.get('pcnf_data_directory', '/var/lib/pgsql'))
-            os.system('sudo -u postgres /usr/bin/pg_basebackup -D {0}/ -Ft -c fast -x -v -P -z'.format(b_dir_temp))
+            pg_version = os.popen('/usr/bin/postmaster --version').read().strip().split(' ')[-1].split('.')
+            if int(pg_version[0]) < 10:
+                os.system('sudo -u postgres /usr/bin/pg_basebackup -D {0}/ -Ft -c fast -x -v -P -z'.format(b_dir_temp))
+            else:
+                os.system('sudo -u postgres /usr/bin/pg_basebackup -D {0}/ -Ft -c fast -X fetch -v -P -z'.format(b_dir_temp))
```

Comment:
This gets a bit unmaintainable. And let's close the handler. While it works, I would refactor it a bit and remove code duplication:

```python
def _get_postgres_version(self):
    '''
    Get postgresql version
    '''
    pg_version = []
    with os.popen('/usr/bin/postmaster --version') as pgh:
        pg_version = map(int, pgh.read().strip().split(' ')[-1].split('.'))

    return pg_version

...

cmd = ['sudo -u postgres /usr/bin/pg_basebackup -D {0}/']
if self._get_postgres_version()[0] < 10:
    cmd.append('-Ft -c fast -x -v -P -z')
else:
    cmd.append('-Ft -c fast -X fetch -v -P -z')
os.system(' '.join(cmd).format(b_dir_temp))
```

WARNING: untested.

Reply:
Ok, so these are purely cosmetic changes. As they need to be tested, I will postpone this and not merge the tested stuff. So hopefully nobody tries to install postgresql10 in the meantime...

Reply:
Well, nothing wrong with the fix on itself, except the code grows and continue copypasting it will be hard to maintain in a future. So I am just asking refactor a bit.